docs/guides/api-gateway/kubernetes.mdx

```diff
@@ -1,430 +1,558 @@
-		<div className="ngrok--card-heading jc-space-between">
-			<h3 className="fw-600">
-				<code>kubectl</code> and Helm
-			</h3>
-		</div>
-		<p>Both installed locally on your workstation.</p>
-	</div>
-</div>
-
-## Why use the K8s Gateway API for your API gateway?
-
-Kubernetes has two APIs for handling ingress from external
-traffic into your cluster's internal network: the original [Ingress
-Controller](https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/)
-or the [Gateway
-API](https://kubernetes.io/docs/concepts/services-networking/gateway/).
-
-The [ngrok Kubernetes Operator](https://github.com/ngrok/ngrok-operator)
-supports both mechanisms. While we encourage you to explore both, we recommend
-the Gateway API based on the following:
-
-- The [Ingress
-  API](https://kubernetes.io/docs/concepts/services-networking/ingress/), which
-  Ingress Controllers utilize, is frozen in new versions of K8s and no longer
-  receives updates.
-- While the Ingress Controller is designed purely for ingress, the Gateway API is
-  also designed for dynamic infrastructure provisioning and advanced traffic routing.
-- The Gateway API is role-oriented, allowing DevOps/platform/infrastructure
-  engineering teams to control clusters, policies, and permissions centrally,
-  while still allowing API developers to self-service endpoints to their services
-  with route objects.
-- The Gateway API provides a common interface to K8s networking, removing
-  reliance on vendor-specific configurations.
-- Many developers can integrate their APIs or apps onto a single `GatewayClass`
-  or `Gateway`, using `HTTPRoute`s, without affecting others.
-
-For the scope of this guide, and based on this recommendation, we'll focus on
-Gateway API.
-
-## **Step 0**. Deploy the example ngrok API service
-
-If you're building a proof of concept with ngrok as the API gateway for your
-APIs in Kubernetes, you can use our [example ngrok API
-service](https://github.com/ngrok-samples/ngrok-api-service). This service
-responds to all routes with information about the connection&mdash;think of it
-as the `pong` to your `curl`.
-
-:::note
-If you already have an API service, you can skip this step&mdash;just be aware
-that you'll need to edit the example configurations below to properly configure
-the ngrok Kubernetes Operator.
-:::
-
-To deploy the ngrok API service to your cluster, apply the [`deployment.yaml`
-file](https://github.com/ngrok-samples/ngrok-api-service/blob/main/kubernetes/deployment.yaml)
-to your cluster, which first creates a new `prod` namespace if one doesn't
-already exist.
-
-```bash
-kubectl apply -f https://raw.githubusercontent.com/ngrok-samples/ngrok-api-service/refs/heads/main/kubernetes/deployment.yaml
+- **[`kubectl`](https://kubernetes.io/docs/tasks/tools/) and [Helm](https://helm.sh/docs/intro/install/)** installed on your local workstation.
+- **A Kubernetes cluster**.
+- **A reserved domain**, which you can get [in the ngrok
```

Comment:
The operator handles domain management for you so you don't need to go to the dashboard first. When you create a resource like `AgentEndpoint`, we create the domain for you if it doesn't already exist.